### PR TITLE
7pm c Search By Content (issue #20)

### PR DESCRIPTION
**Pull request by DanielEskander**
In this pull request, we added a new page for searching through all channels with a given search string. The page will show all the messages containing the search string in a scrollable view, showing the user who sent the message and the contents of the message. At the moment the page is only accessible at the link `member/messages/search`, we will implement another story to add navigation to this page on the Nav bar.

Example:

issue-20

### CODE REVIEW

File: src/main/java/edu/ucsb/mapache/controllers/MessagesController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private ObjectMapper mapper = new ObjectMapper();
+
+    private ResponseEntity<String> getUnauthorizedResponse(String roleRequired) throws JsonProcessingException {
+        Map<String, String> response = new HashMap<String, String>();
+        response.put("error", String.format("Unauthorized; only %s may access this resource.", roleRequired));
+        String body = mapper.writeValueAsString(response);
+        return new ResponseEntity<String>(body, HttpStatus.UNAUTHORIZED);
+    }
+
+    @GetMapping("/contentsearch")
```
Comment (shouc) [inline src/main/java/edu/ucsb/mapache/controllers/MessagesController.java:55]: It seems that the search is not of a specific channel. The name of this method could be misleading.

File: javascript/src/main/components/Channels/ChannelTable.js [hunk 2]
```diff
@@ -1,6 +1,15 @@
 import React from "react";
 import BootstrapTable from 'react-bootstrap-table-next';
 
+const channelLink=(channelName) => {
+    const link = `/member/channels/${channelName}`
+    return (
+
+        < a href= {link} > click here </a>
```
Comment (shouc) [inline javascript/src/main/components/Channels/ChannelTable.js:8]: Shall we say view messages instead here?

**Review by shouc — APPROVED** [review]
Test seems to be failing. not really sure the reason but the rest looks good to me.

**Review by levypik — APPROVED** [review]

**Comment by pconrad** [comment]
@DanielEskander Could you name the PR with a title that says what it does instead of "issue 20"?

**Comment by pconrad** [comment]
@DanielEskander nvm I did it

**Review by pconrad — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
40 points towards team grade awarded

**Pull request merged**